exporter/awsxrayexporter/internal/translator/http.go

```diff
@@ -111,6 +109,12 @@ func makeHTTP(span ptrace.Span) (map[string]pcommon.Value, *awsxray.HTTPData) {
 		return true
 	})
 
+	// if there is no network peer address, then `client.address` is from X-Forwarded-For.
+	_, ok := urlParts[conventions.AttributeNetPeerIP]
```

Comment:
Don't we need to check for `AttributeNetworkPeerAddress`, too?

Reply:
No need. Because there is
```
if !ok && info.Reuqest.ClientIP != nil
```

Reply:
No need, because both `conventions.AttributeNetPeerIP` and  `AttributeNetworkPeerAddress` are stored in`urlParts[conventions.AttributeNetPeerIP]`, which is the pattern we used for compatibility.

Reply:
Adding a boolean flag `hasNetPeerAddr`, so, we can confirm the client IP( only if IP exists) is from X-Forwarded-For or not.
```
if !hasNetPeerAddr && info.Request.ClientIP != nil {
		info.Request.XForwardedFor = aws.Bool(true)
	}
```